wasi/wasi.go

```diff
@@ -101,65 +144,99 @@ type api struct {
 
 func (w *api) register(store *wasm.Store) (err error) {
```

Comment:
while here, let's make the `err = storere.AddHostFunction(...); err != nil {...}` a loop as well

Reply:
sure